mermaid_py/client.py

```diff
@@ -0,0 +1,443 @@
+    samples = [
+        "benthiclittransectmethods",
+        "benthicpittransectmethods",
+        "benthictransects",
+        "beltfishtransectmethods",
+        "bleachingquadratcollectionmethods",
+        "fishbelttransects",
+        "habitatcomplexitytransectmethods",
+        "quadratcollections",
+        "sampleevents",
+        "sampleunitmethods",
+    ]
+
+    def __init__(self, token=None, url=API_DEV_URL):
+        """
+        Constructor for Client base class used for accessing MERMAID API data.
+        :param token: (optional) Authenticated JWT token. Defaults to (token=None).
+        :type token: str
+        :param url: (optional) API URL. Defaults to (url='https://dev-api.datamermaid.org/v1/').
+        :type url: str
+        :return Client class object.
+        """
+        # Instance variables.
+        # MERMAID API URL.
+        self.url = url
+        # JWT token.
+        self.token = token
+        # Authenticated Client instance variable.
+        self.authenticated = False
+        # Checks if a token is provided and assigns authenticated instance variable.
+        if token:
+            self.authenticated = True
+        # Initializes requests Session and assigns headers for all Client MERMAID API calls.
+        self.session = requests.Session()
+        self.session.headers.update(
+            {
+                "content-type": "application/json",
+                "authorization": "Bearer %s" % self.token,
+            }
+        )
+
+    # API paths
+    def _fetch_resource(self, resource=None, parameters=None, data=None):
+        """
+        Prepares API call and utilizes Client Session to make MERMAID API calls.
+        :param resource: (optional) subdirectory path. Defaults to (subdirectory='').
+        :type resource: str
+        :param parameters: (optional) parameters for request.
+        :type parameters: dict:(../?key=val), str:(../?str).
+        :param data: (optional) data. Defaults to (data=None).
+        :return: JSON object containing MERMAID API data.
+        :rtype: dict
+        """
+        # Creates full URL path.
+        if resource is None:
+            raise Exception(f"_fetch_resource invalid resource:{resource}")
+
+        prep_url = self.url + resource
+        # Prepares Request and send from Client class Session.
+        req = requests.Request("GET", url=prep_url, params=parameters, data=data)
+        prepped = self.session.prepare_request(req)
+        resp = self.session.send(prepped)
+
+        # Returns JSON if response code OK.
+        if resp.status_code == requests.codes.ok:
+            return resp.json()
+        else:
+            if resp.status_code == 401:
+                raise Exception(
+                    f"_fetch_resource response code: {resp.status_code} -- Attempt token refresh"
+                )
+
+            else:
+                raise Exception(f"_fetch_resource response code: {resp.status_code}")
+
+    def _api_projects_path(
```

Comment:
Method names are MUCH better... but I would still recommend renaming this, because as is I think it's going to give me back a string path. But it actually fetches data -- so how about something like `_fetch_project_resource`?

(By the way, things like this are easy if you use your IDE's refactoring/rename feature to check actual usages, rather than just doing a search/replace.)